### DISCUSSION TITLE
Suggestion to Integrate Apache Tika to requirement.txt and Expand Transformer Model Support in OpenWebUI

### DISCUSSION THREAD

**Discussion by peuportier** [open-webui/open-webui#5110] [General]
Dear OpenWebUI Team,

I hope this message finds you well. I would like to suggest a couple of enhancements that could significantly improve the user experience and versatility of OpenWebUI, especially for those dealing with large documents.

Integration of Apache Tika: Since Apache Tika is readily available via pip install tika, I propose integrating the Tika library directly into OpenWebUI. By including Tika as part of the default installation, users would have the capability to easily extract and process content from a wide variety of document formats. This functionality could be activated through a simple toggle option in the OpenWebUI admin settings, making it convenient for users who frequently work with diverse document types.

Support for Additional Transformer Models (e.g., LongFormer): I also recommend providing an option to integrate additional transformer models like LongFormer from Hugging Face. This would be particularly beneficial for users dealing with large documents, as models like LongFormer are designed to handle longer contexts more effectively. Currently, the RAG implementation in OpenWebUI seems to focus on the end of documents when recalling information, potentially missing out on important content from the entire document. Offering LongFormer as an alternative could address this issue, allowing for more comprehensive document processing.

Thank you for considering these suggestions. I believe these enhancements would greatly expand the capabilities of OpenWebUI and improve its usefulness for a broader range of applications.

Best regards,
[Aurélien]

**Comment by AlphaCraft9658** [comment]
Great suggestions! I would like to use Tika more easily. Also, I am not sure whether I I have it set up properly right now...

**Comment by roryeckel** [comment]
That Tika python package (https://pypi.org/project/tika/) is a wrapper for the REST API that runs on a server. It does not "contain" the Tika backend. So, this functionality was already implemented in version 0.3.8
Source: https://github.com/open-webui/open-webui/releases/tag/v0.3.8
